crates/worker/src/partition/state_machine/tests/delayed_send.rs

```diff
@@ -0,0 +1,319 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use super::*;
+
+use restate_storage_api::inbox_table::ReadOnlyInboxTable;
+use restate_types::invocation::SubmitNotificationSink;
+use restate_types::time::MillisSinceEpoch;
+use std::time::{Duration, SystemTime};
+use test_log::test;
+
+#[test(tokio::test(flavor = "multi_thread", worker_threads = 2))]
```

Comment:
cc @AhmedSoliman

Reply:
Perhaps related, I think there's some improvements that could be done in general to tests, e.g. let the `test` macro run the test inside the task center (so you don't need to manually set it up at the beginning of each test method)

Reply:
and have some dependency injection a'la junit would be useful too (not sure such thing exists somehow in rust?)

Reply:
I'll do a PR later to improve a bit those tests, and perhaps investigate this multi-thread runtime thingy

Reply:
Can be current_thread. Re. A macro for tc env, I have that planned but it'd be bit tricky to provide something that's flexible enough to let you configure the environment without being too awkward to use. That said, for straight forward cases it should be easy to do. 

As for parametrised tests, it should be easy to do with a simple macro, but in general I'd be cautious about the complexity of tests. I'd love to avoid a test suite that slows us down due to the refactoring needed every time we need to touch something.

Reply:
k will try the current thread thing in the upcoming PR.